src/site/_data/design/components.js

```diff
@@ -284,4 +286,38 @@ module.exports = {
 
     return response;
   },
+
+  // Takes a handle, tries to match that with a component or variant then
+  // applies passed data if available
+  render(handle, data = null) {
+    const item = this.previews.find((item) => item.handle === handle);
+
+    if (!item) {
+      console.log(error(`Component, “${handle}” couldn’t be found`));
+      return '';
+    }
+
+    let renderData = {};
```

Comment:
I think you could simplify this. Yes, using the spread syntax when we only have one object is a copy rather than direct use, but I think all the conditionals here just make it harder to read. Remember spread on `null` or `undefined` is totally valid.

```js
// Passed data overwrites any context data.
const renderData = {...item.data.context, ...data};
return nunjucksEnv.renderString(item.markup, { data: renderData });
```

Reply:
Very tidy!